### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Litvinenko23**

### CODE REVIEW

File: app/test_main.py [hunk 1]
```diff
@@ -1 +1,29 @@
-# write your code here
+from _pytest.monkeypatch import MonkeyPatch
+
+from app.main import can_access_google_page
+
+
+def test_can_access_google_page(monkeypatch: MonkeyPatch) -> None:
+    # Mock the valid_google_url function to always return True
```
Comment (Serhii-Leonenko) [inline app/test_main.py:7]: remove redundant comments

File: app/test_main.py [hunk 2]
```diff
@@ -1 +1,29 @@
...
+    monkeypatch.setattr("app.main.valid_google_url", lambda url: False)
+    assert (
+        can_access_google_page("https://www.invalidurl.com")
+        == "Not accessible"
+    )
+
+    # Test with a valid URL but no internet connection
+    monkeypatch.setattr("app.main.valid_google_url", lambda url: True)
+    monkeypatch.setattr("app.main.has_internet_connection", lambda: False)
+    assert can_access_google_page("https://www.google.com") == "Not accessible"
```
Comment (Serhii-Leonenko) [inline app/test_main.py:29]: add tests for case when both are False, also try to remove code duplicates

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]
Use @pytest.mark.parametrize for DRY

File: app/test_main.py [hunk 3]
```diff
@@ -1 +1,26 @@
...
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, is_valid, has_connection, expected",
+    [
+        ("https://www.google.com", True, True, "Accessible"),
+        ("https://www.goo!gle.com", False, True, "Not accessible"),
+        ("https://www.google.com", True, False, "Not accessible"),
+        ("https://www.goole.com", False, False, "Not accessible"),
```
Comment (wQuelS) [inline app/test_main.py:12]: Provided more descriptive message

File: app/test_main.py [hunk 4]
```diff
@@ -1 +1,26 @@
-# write your code here
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+from app.main import can_access_google_page
```
Comment (wQuelS) [inline app/test_main.py:3]: make your imports according to PEP8 convention. Blank lines between group of imports, their order, etc.

**Review by wQuelS — CHANGES_REQUESTED** [review]
